packages/protocol-sdk/src/anvil.ts

```diff
@@ -0,0 +1,84 @@
+import { spawn } from "node:child_process";
+import { join } from "path";
+import { test } from "vitest";
+import {
+  PublicClient,
+  TestClient,
+  WalletClient,
+  createPublicClient,
+  createTestClient,
+  createWalletClient,
+  http,
+} from "viem";
+import { foundry } from "viem/chains";
+
+export interface AnvilViemClientsTest {
+  viemClients: {
+    walletClient: WalletClient;
+    publicClient: PublicClient;
+    testClient: TestClient;
+  };
+}
+
+async function waitForAnvilInit(anvil: any) {
+  return new Promise((resolve) => {
+    anvil.stdout.once("data", () => {
+      resolve(true);
+    });
+  });
+}
+
+export const anvilTest = test.extend<AnvilViemClientsTest>({
+  viemClients: async ({task}, use) => {
+    console.log('setting up clients for ', task.name);
+    const port = Math.floor(Math.random() * 2000) + 4000;
+    const anvil = spawn(
+      "anvil",
+      [
+        "--port",
+        `${port}`,
+        "--fork-url",
+        "https://rpc.zora.co/",
+        "--fork-block-number",
+        "6133407",
```

Comment:
how about allowing chain name (`zora`, or `zora-sepolia`) and `blockNumber` to be specified as arguments with defaults?